lighthouse-core/test/config/config-test.js

```diff
@@ -384,39 +442,61 @@ describe('Config', () => {
       assert.equal(config.passes.length, 2, 'incorrect # of passes');
       assert.ok(config.audits.length < totalAuditCount, 'audit filtering probably failed');
     });
 
     it('should filter out other passes if passed PWA', () => {
       const totalAuditCount = origConfig.audits.length;
-      const config = Config.generateNewConfigOfCategories(origConfig, ['pwa']);
+      const config = Config.generateNewFilteredConfig(origConfig, ['pwa']);
       assert.equal(Object.keys(config.categories).length, 1, 'other categories are present');
       assert.ok(config.audits.length < totalAuditCount, 'audit filtering probably failed');
     });
 
     it('should filter out other passes if passed Best Practices', () => {
       const totalAuditCount = origConfig.audits.length;
-      const config = Config.generateNewConfigOfCategories(origConfig, ['best-practices']);
+      const config = Config.generateNewFilteredConfig(origConfig, ['best-practices']);
       assert.equal(Object.keys(config.categories).length, 1, 'other categories are present');
       assert.equal(config.passes.length, 2, 'incorrect # of passes');
       assert.ok(config.audits.length < totalAuditCount, 'audit filtering probably failed');
     });
 
     it('should only run audits for ones named by the category', () => {
-      const config = Config.generateNewConfigOfCategories(origConfig, ['performance']);
+      const config = Config.generateNewFilteredConfig(origConfig, ['performance']);
       const selectedCategory = origConfig.categories.performance;
       const auditCount = Object.keys(selectedCategory.audits).length;
 
       assert.equal(config.audits.length, auditCount, '# of audits match aggregation list');
     });
+
+    it('should only run specified audits', () => {
+      const config = Config.generateNewFilteredConfig(origConfig, [], ['works-offline']);
+      assert.equal(config.passes.length, 2, 'incorrect # of passes');
+      assert.equal(config.audits.length, 1, 'audit filtering failed');
+    });
+
+    it('should combine audits and categories additively', () => {
+      const config = Config.generateNewFilteredConfig(origConfig, ['performance'], ['is-on-https']);
+      const selectedCategory = origConfig.categories.performance;
+      const auditCount = Object.keys(selectedCategory.audits).length + 1;
+      assert.equal(config.passes.length, 2, 'incorrect # of passes');
+      assert.equal(config.audits.length, auditCount, 'audit filtering failed');
+    });
+
+    it('should support redundant filtering', () => {
+      const config = Config.generateNewFilteredConfig(origConfig, ['pwa'], ['is-on-https']);
+      const selectedCategory = origConfig.categories.pwa;
+      const auditCount = Object.keys(selectedCategory.audits).length;
+      assert.equal(config.passes.length, 3, 'incorrect # of passes');
+      assert.equal(config.audits.length, auditCount, 'audit filtering failed');
+    });
```

Comment:
tests for the new warnings would be good, too. `log.events.addListener('warning', callback)` can be an easy way to do that rather than going to the trouble of overriding `log`, like is done [elsewhere](https://github.com/GoogleChrome/lighthouse/blob/605a4579c5cd0eeba9ea71a3dcf38d7817d1b86a/lighthouse-core/test/config/config-test.js#L88-L95) in that file